cli/flox-watchdog/src/logger.rs

```diff
@@ -40,3 +46,25 @@ pub(crate) fn init_logger(file_path: &Option<PathBuf>) -> Result<(), anyhow::Err
         .init();
     Ok(())
 }
+
+/// Emits a log entry at an interval which is used as an indication of whether a
+/// watchdog's log file can be garbage collected.
+pub(crate) async fn log_heartbeat() -> Sender<()> {
+    let (stop_tx, mut stop_rx) = watch::channel(());
+    let mut interval = interval(HEARTBEAT_INTERVAL);
+
+    tokio::spawn(async move {
+        loop {
+            tokio::select! {
+                _ = interval.tick() => {
+                    debug!("still watching, woof woof");
+                }
+                _ = stop_rx.changed() => {
+                    break;
+                }
+            }
+        }
+    });
+
+    stop_tx
+}
```

Comment:
suggestion: use sync threading
Introducing tokio for this is a bit heavy IMO, considering that tokio will spawn an entire threadpool.
Also using just mpsc we can achieve something similar:


```rust
/// Emits a log entry at an interval which is used as an indication of whether a
/// watchdog's log file can be garbage collected.
pub(crate) async fn log_heartbeat() -> Sender<()> {
	// could probably also a bounded sync_channel
	// and benefit from a marker type `struct Exit`
    // technically just dropping the sender would be enough of a signal
    let (stop_tx, mut stop_rx) = mpsc::channel::<()>(); 
   
    thread::spawn(async move {
        loop {
           match stop_rx.receive_timeout(HEARTBEAT_INTERVAL) { 
                Err(mpsc::RecvTimeoutError::Timeout) => {
                    debug!("still watching, woof woof");
                },
                Ok(_) | Err(_) => {
                    // exit signal received, or sender gone, bye
                    break
           		},
           }
        }
    });

    stop_tx
}
```

Reply:
Rebased into b1d8770:

- `std::thread::spawn` instead of `tokio`
- simple loop with no cancellation that will die with the main thread
- renamed the function and update the comment to reflect changes